src/backend/pipeline/trigger/triggerfuncs.c

```diff
@@ -0,0 +1,213 @@
+/*-------------------------------------------------------------------------
+ *
+ * triggerfuncs.c
+ *	  Functions to call from continuous triggers
+ *
+ * Copyright (c) 2013-2015, PipelineDB
+ *
+ *-------------------------------------------------------------------------
+ */
+#include "pipeline/trigger/triggerfuncs.h"
+#include "storage/shmem.h"
+#include "lib/stringinfo.h"
+#include "commands/trigger.h"
+#include "utils/rel.h"
+#include "nodes/print.h"
+#include "pipeline/trigger/trigger.h"
+#include "pipeline/trigger/tuple_formatter.h"
+#include "pipeline/trigger/config.h"
+
+#include <fcntl.h>
+#include <stdio.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <sys/uio.h>
+#include <unistd.h>
+#include "utils/builtins.h"
+#include "miscadmin.h"
+
+static int fd = -1;
+static const char *log_file_name = "/tmp/.pipelinedb_cluster_test.log";
+
+static void check_file()
+{
+	if (fd == -1)
+	{
+		fd = open(log_file_name, O_RDWR | O_APPEND, 0600);
+		Assert(fd != -1);
+	}
+}
+
+static void
+write_to_file(StringInfo info)
+{
+	struct iovec iov;
+	check_file();
+
+	appendStringInfoChar(info, '\n');
+
+	iov.iov_base = info->data;
+	iov.iov_len = info->len;
+
+	/* Use writev to atomically write the tuple to the file */
+
+	writev(fd, &iov, 1);
+}
+
+/*
+ * send_alert_new_row
+ *
+ * Converts the 'new' tuple to copy format, and sends to the alert server
+ */
+PG_FUNCTION_INFO_V1(pipeline_send_alert_new_row);
```

Comment:
I don't thing we need this declaration here. These functions don't need to be part of the `pg_proc` infrastructure. They should be private functions for the trigger proc.